src/components/AttendantForm.jsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from "react";
+import { getJobTitles, addAttendant, getAttendants } from "../api";
+import AttendantDisplay from "./AttendantDisplay";
+
+export default function AttendantForm() {
+  const [attendants, setAttendants] = useState([]);
+  const [name, setName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [jobTitle, setJobTitle] = useState("");
+  const [age, setAge] = useState("");
+  const [allJobTitles, setAllJobTitles] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function submitHandler(e) {
+    e.preventDefault();
+    const submittedData = {
+      name,
+      lastName,
+      jobTitle,
+      age,
+    };
+
+    try {
+      setIsLoading(true);
+      const { status } = await addAttendant(submittedData);
+      if (status === 200) {
+        const { status: attendancesStatus, data } = await getAttendants();
+        if (attendancesStatus === 200) {
+          setAttendants([...data].sort((a, b) => a.age - b.age));
+        }
+      }
+    } catch (e) {
+      console.error("failed to add attendant", e);
+    } finally {
+      clearForm();
+      setIsLoading(false);
+    }
+  }
+
+  function clearForm() {
+    [setName, setLastName, setJobTitle, setAge].forEach((setter) => setter(""));
+  }
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    Promise.all([getJobTitles(), getAttendants()])
+      .then(([jobTitles, initialAttendants]) => {
+        if (jobTitles.status === 200) {
+          setAllJobTitles([...jobTitles.data].sort());
+        }
+        if (initialAttendants.status === 200) {
+          setAttendants(
+            [...initialAttendants.data].sort((a, b) => a.age - b.age)
+          );
+        }
+      })
+      .catch((e) => {
+        console.error("failed to get api response", e);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  return (
+    <div>
+      <hr />
+      <h1>Attendant Registration Form</h1>
+      <br />
+      <form
+        className="attendant-form"
+        onSubmit={submitHandler}
+      >
+        <div>
+          <label htmlFor="name">Enter name:</label>
+          <input
+            type="text"
+            minLength="2"
+            maxLength="50"
+            pattern="[A-Za-zÀ-ÖØ-öø-ÿ'\-]+"
+            name="name"
+            placeholder="Name"
+            data-testid="name"
+            value={name}
+            required
+            onChange={(e) => {
+              setName(e.target.value);
+            }}
```

Comment:
The practice is to have a separate `handleSomethingChange` function that is passed to the `onChange` prop. Same goes for all html elements below
```suggestion
            onChange={handleNameChange}
```